dot/core/service_integration_test.go

```diff
@@ -361,97 +454,70 @@ func TestHandleChainReorg_WithReorg_Transactions(t *testing.T) {
 }
 
 func TestMaintainTransactionPool_EmptyBlock(t *testing.T) {
-	// TODO: update these to real extrinsics on update to v0.9 (#904)
-	txs := []*transaction.ValidTransaction{
-		{
-			Extrinsic: []byte("a"),
-			Validity:  &transaction.Validity{Priority: 1},
-		},
-		{
-			Extrinsic: []byte("b"),
-			Validity:  &transaction.Validity{Priority: 4},
-		},
-		{
-			Extrinsic: []byte("c"),
-			Validity:  &transaction.Validity{Priority: 2},
-		},
-		{
-			Extrinsic: []byte("d"),
-			Validity:  &transaction.Validity{Priority: 17},
-		},
-		{
-			Extrinsic: []byte("e"),
-			Validity:  &transaction.Validity{Priority: 2},
-		},
+	encExt, rt := generateTestValidTxns(t)
+	cfg := &Config{
+		Runtime: rt,
 	}
 
 	ctrl := gomock.NewController(t)
 	telemetryMock := NewMockClient(ctrl)
 	telemetryMock.EXPECT().SendMessage(gomock.Any()).AnyTimes()
 
 	ts := state.NewTransactionState(telemetryMock)
-	hashes := make([]common.Hash, len(txs))
-
-	for i, tx := range txs {
-		h := ts.AddToPool(tx)
-		hashes[i] = h
+	tx := &transaction.ValidTransaction{
+		Extrinsic: types.Extrinsic(encExt),
+		Validity:  &transaction.Validity{Priority: 1},
 	}
+	_ = ts.AddToPool(tx)
 
-	s := &Service{
-		transactionState: ts,
+	s := NewTestService(t, cfg)
+	s.transactionState = ts
+
+	provides := common.MustHexToBytes("0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d00000000")
```

Comment:
what is this? the previous tx hash?

Reply:
so you are expecting this `Provides` tags to be added via the runtime? I would put a comment about it.  It would be nice to have a link to the runtime code that appends these tags, and what these tags are.

Reply:
I believe so? Haven't been able to find much on it. In substrate the provides is a vector of transaction tags, which is just a vector of bytes. Haven't been able to find that the possible values. My understanding is that the provides tags are added by the runtime to deal with transaction dependencies. So maybe it is the previous txn hash @noot? Or at least A previous transaction of which the current txn depends?

Reply:
Will try this now, just getting to this round of feedback

Reply:
It seems that this test does require the Provides param to pass. @noot correct me if im wrong but it seems to be added by the runtime. When i remove it a check fails because its getting set somewhere, but I haven't been able to find where its set.

Unless told otherwise, thinking I will leave as is and add a comment

Reply:
weird, ok feel free to leave it and just document it then